docs/docs/architecture.md

```diff
@@ -1,143 +1,148 @@
 # Architecture overview
 
-If you are new to the concept of Confidential Computing, you may also find the [introduction to enclaves](enclaves.md)
-document helpful.
+This page covers the basic architecture of Conclave, a [Confidential Computing](enclaves.md) platform.
 
 ## Primary entities
 
-There are three entities in an application that uses Conclave:
+There are three main entities in any Conclave application.
 
-1. Clients
+1. Enclaves
 2. Hosts
-3. Enclaves
+3. Clients
 
-**Clients** send and receive encrypted messages to/from enclaves by interacting with the host over the network.
-Conclave doesn't mandate any particular network protocol for client<->host communication. It's up to you. However
-the content of the messages *is* defined, using the [Mail](#mail) API. Mail is described below.
+**Enclaves** are protected regions of memory in which you can write the confidential part of an application. An 
+enclave resides inside a host program. However, a host or its owner cannot view or modify the code running in an 
+enclave.
 
-**Host programs** load enclaves. From a security perspective they are fully untrusted and assumed to be malicious at all
-times. Hosts are relied on to provide the enclave with resources but beyond that work only with encrypted data. In some
-kinds of app their function is primarily proxying communications from clients to enclaves, but sometimes they also assist
-with application logic. Hosts use a standard JVM like HotSpot.
+**Hosts** are programs that load enclaves. They are untrusted and assumed to be malicious at all times. Hosts use a 
+standard JVM like HotSpot. The host JVM and the enclave run inside the same operating system process. Clients 
+communicate with enclaves through host programs. In some applications, hosts also assist with the application logic. 
 
-**Enclaves** are classes that are loaded into a dedicated sub-JVM with a protected memory space, running inside the same
-operating system process as the host JVM. Code running in an enclave cannot be tampered with by the host system or its owner,
-nor can the host system or its owner see the data that the enclave is processing. Enclaves communicate with clients via 
-the host program.
+**Clients** are programs that send and receive encrypted messages through hosts to communicate with enclaves. In a 
```

Comment:
Clients are the remote applications that communicate with the enclave via the host, by sending and receiving encrypted messages with it.

Reply:
done.